modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/QueryMemoryManager.java

```diff
@@ -239,7 +226,18 @@ public boolean onQuotaExceeded(long size) {
      * @param newGlobalQuota New global query quota.
      */
     public synchronized void setGlobalQuota(String newGlobalQuota) {
-        this.globalQuota = U.parseBytes(newGlobalQuota);
+        long globalQuota0 = U.parseBytes(newGlobalQuota);
+        long heapSize = Runtime.getRuntime().maxMemory();
+
+        A.ensure(
+            heapSize > globalQuota0,
+            "Sql global memory quota can't be more than heap size: heapSize="
+                + heapSize + ", quotaSize=" + globalQuota0
+        );
+
+        A.ensure(globalQuota0 >= 0, "Sql global memory quota must be >= 0: quotaSize=" + globalQuota0);
```

Comment:
Is it ok?
Negative values were allowed to disable global quota.

Reply:
according to IgniteConfiguration#setSqlGlobalMemoryQuota 0 means no quota